<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Player</class>
 <widget class="QWidget" name="Player">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>958</width>
    <height>114</height>
   </rect>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <property name="windowIcon">
   <iconset resource="../../resources.qrc">
    <normaloff>:/images/icon.ico</normaloff>:/images/icon.ico</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="controlPogressLayout">
     <item row="1" column="0">
      <layout class="QHBoxLayout" name="controlLayout">
       <item>
        <widget class="QPushButton" name="shuffleButton">
         <property name="toolTip">
          <string>Shuffle (R)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/shuffle.png</normaloff>:/images/player/shuffle.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>R</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="repeatButton">
         <property name="toolTip">
          <string>Repeat (Ctrl + T)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/repeat.png</normaloff>:/images/player/repeat.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+T</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="leftSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="advanceBack">
         <property name="toolTip">
          <string>Previous Track (Up Arrow)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/advance-prev.png</normaloff>:/images/player/advance-prev.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Up</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="leftNextSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="replayButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Replay by 10 seconds (Left Arrow)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/replay-10.png</normaloff>:/images/player/replay-10.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Left</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="playbackButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Play (Space)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/pause.png</normaloff>:/images/player/pause.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>48</width>
           <height>48</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Space</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="default">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="forwardButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Forward by 10 seconds (Right Arrow)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/forward-10.png</normaloff>:/images/player/forward-10.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Right</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="rightNextSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="advanceNext">
         <property name="toolTip">
          <string>Next Track (Down Arrow)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/advance-next.png</normaloff>:/images/player/advance-next.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Down</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="rightSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="stopButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="toolTip">
          <string>Stop (S)</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../../resources.qrc">
           <normaloff>:/images/player/stop.png</normaloff>:/images/player/stop.png</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>S</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="speedButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <family>Digital-7</family>
           <pointsize>21</pointsize>
          </font>
         </property>
         <property name="toolTip">
          <string>Adjust Playback Speed (Ctrl + P)</string>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LayoutDirection::LeftToRight</enum>
         </property>
         <property name="text">
          <string>1.00</string>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+P</string>
         </property>
         <property name="autoRepeat">
          <bool>true</bool>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
         <property name="default">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="progressLayout">
       <item>
        <widget class="QLabel" name="elapsedTimeLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>00:00</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="progressSlider">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="totalTimeLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>00:00</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>playbackButton</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>changePlaybackMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>376</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>replayButton</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>replay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>373</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>283</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>forwardButton</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>forward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>800</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>669</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speedButton</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>changePlaybackSpeed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>852</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>775</x>
     <y>6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>repeatButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>Player</receiver>
   <slot>setInfiniteMode(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shuffleButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>Player</receiver>
   <slot>setShuffleMode(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>35</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advanceBack</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>goToPreviousTrack()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advanceNext</sender>
   <signal>clicked()</signal>
   <receiver>Player</receiver>
   <slot>goToNextTrack()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>810</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>shuffleModeChnaged(bool)</signal>
  <signal>infiniteModeChanged(bool)</signal>
  <slot>changePlaybackMode()</slot>
  <slot>stop()</slot>
  <slot>replay()</slot>
  <slot>forward()</slot>
  <slot>controlPlaylist()</slot>
  <slot>changePlaybackSpeed()</slot>
  <slot>setInfiniteMode(bool)</slot>
  <slot>updateInfiniteButton()</slot>
  <slot>setShuffleMode(bool)</slot>
  <slot>updateShuffleButton()</slot>
  <slot>goToNextTrack()</slot>
  <slot>goToPreviousTrack()</slot>
 </slots>
</ui>
